jQuery datpicker()

shumoo opened the pull request:
Replaced drop down menu from discussion with a datepicker input field. Added jquery-ui.js which should probably be thinned out or included differently.  I didn't want to delay putting this up since RJ was kind enough to help me.  I'll fix whatever needs to be updated.

shumoo committed:
jQuery datpicker()

Replaced drop down menu from discussion with a datepicker input field.

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
```
There is a version of jquery-ui.js shipped with Vanilla. You should use that in order to avoid version conflicts.

shumoo replied:
completed this...will check in with the rest of the updates. and do another pull request.

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
```
I'm not sure what the best way would be: adding the css like you did, use the css from a CDN or just add the markup needed for the calendar to the custom css.
From all that, using a CDN would be my favorite...

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+         $Hidden = ' Hidden';   
+      }
+	  
+	  //Removed when switching drop down fields to a date picker
```
If you manage to create the pull request so that the files are comparable, you could safely delete instead of commenting out ;)

shumoo replied:
old habbits...

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+         return;
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);	  
```
You should stick to the naming conventions. So better use $Time instead of $time. Shouldn't it be better $Date anyway?

shumoo replied:
I guess so, reading up on the datatype being time made me think it would be appropriate but I can see your perspective too.

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);	  
+	  $formatted_date = date('Y-m-d',$time);	  
```
See above: $FormattedDate

R-J commented on class.eventcalendar.plugin.php:
```diff
@@ -1,265 +1,279 @@
+      if(!CheckPermission(array('Plugins.EventCalendar.View'))) {
+         return;
+      }
+      if ($EventDate != '0000-00-00' & !is_null($EventDate))  {
```
I never officially learned PHP and that's why I'm not ashamed to ask: mustn't it be "&&" instead of "&"?
But I would prefer to change the function to `private function FormatEventCalendarDate($EventDate = '0000-00-00' , $IncludeIcon = FALSE) {` to avoid the null check

R-J commented on design/eventcalendar.css:
```diff
@@ -49,11 +49,11 @@
 li.Sunday {
    background-color: #CFECFF;
-   color: red;
+   color: #red;
```
That is wrong. It must really be `color: red`

R-J commented on design/eventcalendar.css:
```diff
@@ -49,11 +49,11 @@
 li.Sunday > a {
    /* background-color: #CFECFF; */
-   color: red;
+   color: #red;
```
See above.

R-J commented:
I have commented the changes above. My biggest concern is that the changes in the plugin couldn't be identified. I made a diff with a text edit which was quite painful if you take into consideration that this should be an advantage of using git ;)

The files `views/month.php` and `js/eventcalendar.js` are not changed at all.

shumoo committed:
using js included in vanilla

dropping the lib I used.

shumoo committed:
Updates for js, formatting, and bug

removed jquery lib, linked to included jquery-ui js file.  formatted
variable to follow naming convention.  corrected logical operator (& to
&&).

shumoo committed:
fixed git

should look nice to compare now

shumoo commented:
I made multiple updates per R-J's comments.  The only thing that remains unchanged from the original pull is that I have not changed the new CSS file and how it's being imported; I don't know how to go about getting or creating a CDN at the moment.

shumoo committed:
adding png file with images for buttons etc...

forgot to add this originally, needed for buttons etc....

shumoo committed:
Added Day of the week Name

Added Day of the week Name to the calendar...people were confused that
the first day of the week displayed wasn't Sunday.

shumoo commented:
that was easier than I thought it might be...
